### PR TITLE
local operator catalog raw file based format

### PR DESCRIPTION
**Pull request by eguzki**
### what

* Locally built custom OLM catalogs use (raw) File Based Catalog format instead of the deprecated SQLITE format.
From [SQLite-based catalogs](https://docs.openshift.com/container-platform/4.9/operators/admin/olm-managing-custom-catalogs.html#olm-managing-custom-catalogs-sqlite),

```
The SQLite database format for Operator catalogs is a deprecated feature. Deprecated functionality is still included in OpenShift Container Platform and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments.
```

* Update github actions to push file based catalog format images
* Makefile
  * target renamed: `deploy-olm` -> `deploy-catalog`
  * target renamed: `undeploy-olm` -> `undeploy-catalog`
  * function renamed: `go-get-tool` -> `go-install-tool`
  * `bundle-dependencies` target removed. Functionality moved to the `bundle` target.
  * replace `envsubst` operations to update yaml objects with external env variarbles with the `yq` tool
  * moved all catalog related logic to `make/catalog.mk`
  * New: `catalog` target to build catalog manifests
* Documentation oriented to contributors `doc/development.md`:
  * New section: `Deploy kuadrant operator using OLM`
  * New section: "Build custom OLM catalog`
  * Added Table of Contents.

TODO

- [x] Doc
- [x] github actions

### Verification steps

Build local catalog. It will include kuadrant operator, limitador operator and authorino operator

```
make catalog
make catalog-build CATALOG_IMG=quay.io/kuadrant/kuadrant-operator-catalog:local-testing
```

Push the image to Quay.io. Currently it is required because the catalogsource sets the image pull policy to `Always`, so even if the image is loaded in the kind cluster, it will fail if the image is not in quay.

```
make catalog-push CATALOG_IMG=quay.io/kuadrant/kuadrant-operator-catalog:local-testing
```

Create kind cluster

```
make kind-create-cluster
```

Deploy OLM

```
make install-olm
```

Deploy kuadrant operator using OLM

```
make deploy-catalog CATALOG_IMG=quay.io/kuadrant/kuadrant-operator-catalog:local-testing
```

Check the kuadrant operator, limitador operator and authorino operator are up and running

```
k get pods -n kuadrant-system
NAME                                                              READY   STATUS      RESTARTS   AGE
43539917290efc2e6f33b021e25da1f7a9f150d8d276acf92fd09b--1-wqggr   0/1     Completed   0          34s
6427bb5aac2edc4750e505a33be4eb4b56923d3aed4213abecdc70--1-v4lk5   0/1     Completed   0          34s
aecaf21d9bde4883921cefdd0ba3a142f079e897473c5d40fd8a1a--1-2249v   0/1     Completed   0          34s
authorino-operator-7d9968586c-x4kbl                               1/1     Running     0          16s
kuadrant-operator-catalog-rkmdn                                   1/1     Running     0          79s
kuadrant-operator-controller-manager-6688b8bfdf-m4rwh             1/1     Running     0          13s
limitador-operator-controller-manager-7f5dcdd976-66bcs            1/1     Running     0          11s
```

### CODE REVIEW

**Comment by eguzki** [comment]
sorry for the bit long PR :pray:

**Review by alexsnaps — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -131,26 +123,69 @@ all: build
...
 KUSTOMIZE = $(PROJECT_PATH)/bin/kustomize
-$(KUSTOMIZE): 
-	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.5.5)
+$(KUSTOMIZE):
+	$(call go-install-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v4@v4.5.5)
 
+.PHONY: kustomize
 kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
 
+YQ=$(PROJECT_PATH)/bin/yq
```
Comment (didierofrivia) [inline Makefile:150]: 💄

File: doc/development.md [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+   * [Run locally](#run-locally)
+   * [Deploy the operator in a deployment object](#deploy-the-operator-in-a-deployment-object)
+   * [Deploy kuadrant operator using OLM](#deploy-kuadrant-operator-using-olm)
+   * [Build custom OLM catalog](#build-custom-olm-catalog)
+      * [Build kuadrant operator bundle image](#build-kuadrant-operator-bundle-image)
+      * [Build custom catalog](#build-custom-catalog)
+      * [Deploy kuadrant operator using OLM](#deploy-kuadrant-operator-using-olm-1)
+   * [Cleaning up](#cleaning-up)
+   * [Run tests](#run-tests)
+      * [Unittests](#unittests)
```
Comment (didierofrivia) [inline doc/development.md:15]:
```suggestion
      * [Unit tests](#unittests)
```

File: doc/development.md [hunk 3]
```diff
@@ -43,6 +65,145 @@ List of tasks done by the command above:
...
+Authorino bundles. These are the example commands to build the manifests, build the bundle image
+and push to the regitry.
+
+In the example, a new kuadrant operator bundle version `0.8.0` will be created that references
+the kuadrant operator image `quay.io/kuadrant/kuadrant-operator:v0.5.0` and latest Limitador and
+Authorino bundles.
+
+```bash
+# manifests
+make bundle IMG=quay.io/kuadrant/kuadrant-operator:v0.5.0 VERSION=0.8.0
```
Comment (didierofrivia) [inline doc/development.md:159]:
```suggestion
make manifests IMG=quay.io/kuadrant/kuadrant-operator:v0.5.0 VERSION=0.8.0
```

**Comment by didierofrivia** [comment]
It looks good, minor things I commented. However, I've still haven't successfully tried it out, having network issues:

```
dd@pomodoro:~/Projects/RedHat/kuadrant-operator|local-catalog-file-based⚡ ⇒  make install-olm
./utils/install-operator-sdk.sh /Users/dd/Projects/RedHat/kuadrant-operator/bin/operator-sdk v1.22.0
Downloading https://github.com/operator-framework/operator-sdk/releases/download/v1.22.0/operator-sdk_darwin_arm64
gpg: connecting dirmngr at '/Users/dd/.gnupg/S.dirmngr' failed: IPC connect call failed
gpg: keyserver receive failed: No dirmngr
make: *** [/Users/dd/Projects/RedHat/kuadrant-operator/bin/operator-sdk] Error 2
dd@pomodoro:~/Projects/RedHat/kuadrant-operator|local-catalog-file-based⚡ ⇒  make install-olm
./utils/install-operator-sdk.sh /Users/dd/Projects/RedHat/kuadrant-operator/bin/operator-sdk v1.22.0
Downloading https://github.com/operator-framework/operator-sdk/releases/download/v1.22.0/operator-sdk_darwin_arm64
make: *** [/Users/dd/Projects/RedHat/kuadrant-operator/bin/operator-sdk] Error 92
dd@pomodoro:~/Projects/RedHat/kuadrant-operator|local-catalog-file-based⚡ ⇒  make install-olm
./utils/install-operator-sdk.sh /Users/dd/Projects/RedHat/kuadrant-operator/bin/operator-sdk v1.22.0
Downloading https://github.com/operator-framework/operator-sdk/releases/download/v1.22.0/operator-sdk_darwin_arm64
gpg: key 052996E2A20B5C7E: "Operator SDK (release) <cncf-operator-sdk@cncf.io>" 2 new signatures
gpg: Total number processed: 1
gpg:         new signatures: 2
gpg: assuming signed data in 'checksums.txt'
gpg: Signature made Tue Jun 14 22:57:45 2022 CEST
gpg:                using RSA key 8613DB87A5BA825EF3FD0EBE2A859D08BF9886DB
gpg: Good signature from "Operator SDK (release) <cncf-operator-sdk@cncf.io>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 3B2F 1481 D146 2380 80B3  46BB 0529 96E2 A20B 5C7E
     Subkey fingerprint: 8613 DB87 A5BA 825E F3FD  0EBE 2A85 9D08 BF98 86DB
operator-sdk_darwin_arm64: OK
/Users/dd/Projects/RedHat/kuadrant-operator/bin/operator-sdk olm install
INFO[0000] Fetching CRDs for version "latest"
INFO[0000] Fetching resources for resolved version "latest"
I1207 18:09:07.865914   53853 request.go:601] Waited for 1.047390292s due to client-side throttling, not priority and fairness, request: GET:https://127.0.0.1:61648/apis/certificates.k8s.io/v1?timeout=32s
INFO[0018] Creating CRDs and resources
INFO[0018]   Creating CustomResourceDefinition "catalogsources.operators.coreos.com"
INFO[0018]   Creating CustomResourceDefinition "clusterserviceversions.operators.coreos.com"
INFO[0018]   Creating CustomResourceDefinition "installplans.operators.coreos.com"
INFO[0018]   Creating CustomResourceDefinition "olmconfigs.operators.coreos.com"
INFO[0018]   Creating CustomResourceDefinition "operatorconditions.operators.coreos.com"
INFO[0018]   Creating CustomResourceDefinition "operatorgroups.operators.coreos.com"
INFO[0018]   Creating CustomResourceDefinition "operators.operators.coreos.com"
INFO[0018]   Creating CustomResourceDefinition "subscriptions.operators.coreos.com"
INFO[0018]   Creating Namespace "olm"
INFO[0018]   Creating Namespace "operators"
INFO[0018]   Creating ServiceAccount "olm/olm-operator-serviceaccount"
INFO[0018]   Creating ClusterRole "system:controller:operator-lifecycle-manager"
INFO[0018]   Creating ClusterRoleBinding "olm-operator-binding-olm"
INFO[0018]   Creating OLMConfig "cluster"
INFO[0022]   Creating Deployment "olm/olm-operator"
INFO[0022]   Creating Deployment "olm/catalog-operator"
INFO[0022]   Creating ClusterRole "aggregate-olm-edit"
INFO[0022]   Creating ClusterRole "aggregate-olm-view"
INFO[0022]   Creating OperatorGroup "operators/global-operators"
INFO[0022]   Creating OperatorGroup "olm/olm-operators"
INFO[0022]   Creating ClusterServiceVersion "olm/packageserver"
INFO[0022]   Creating CatalogSource "olm/operatorhubio-catalog"
INFO[0022] Waiting for deployment/olm-operator rollout to complete
INFO[0022]   Waiting for Deployment "olm/olm-operator" to rollout: 0 of 1 updated replicas are available
FATA[0120] Failed to install OLM version "latest": deployment/olm-operator failed to rollout: timed out waiting for the condition
```

**Comment by didierofrivia** [comment]
Ok, finally managed to try it out and it works like a charm!

```bash
dd@pomodoro:~/Projects/RedHat/kuadrant-operator|local-catalog-file-based⚡ ⇒  k get pods -n kuadrant-system
NAME                                                              READY   STATUS      RESTARTS   AGE
43539917290efc2e6f33b021e25da1f7a9f150d8d276acf92fd09b--1-f9dvv   0/1     Completed   0          4m57s
6427bb5aac2edc4750e505a33be4eb4b56923d3aed4213abecdc70--1-rjxdt   0/1     Completed   0          4m57s
aecaf21d9bde4883921cefdd0ba3a142f079e897473c5d40fd8a1a--1-rsdzb   0/1     Completed   0          4m57s
authorino-operator-7d9968586c-r658g                               1/1     Running     0          3m38s
kuadrant-operator-catalog-krz9t                                   1/1     Running     0          5m40s
kuadrant-operator-controller-manager-6688b8bfdf-h9h6g             1/1     Running     0          3m36s
limitador-operator-controller-manager-7f5dcdd976-zcl5w            1/1     Running     0          3m41s
```

**Review by didierofrivia — APPROVED** [review]

**Pull request merged**